GUVNOR-2308: Project explorer minimize/maximize button overlaps with breadcrumbs

paulovmr opened the pull request:
When the project explorer is wide, it looks like that:
![image1](https://cloud.githubusercontent.com/assets/832830/12481229/d14c2a3c-c02e-11e5-837e-b44ceb45230a.png)

and when it is narrow, it looks like that:
![image2](https://cloud.githubusercontent.com/assets/832830/12481232/d69c32de-c02e-11e5-8350-f1f7adb3078a.png)

manstis commented:
(Unrelated) the drop downs in the breadcrumbs look wrong - they have an icon "down arrow" and what appears to be a lower case "v".. we should only have the icon IIRC.

paulovmr commented:
@manstis Yes, I don't know why but since I updated my repos this morning the workbench style became a little different (which includes the dropdowns).

manstis commented:
Is it possible you can follow that up with Cristiano? He may have some idea
what changed in Uberfire.

paulovmr commented:
Sure, I will talk to him (and investigate if this is only happening in my environment).

paulovmr commented:
jenkins, retest this

jhrcek commented:
@manstis @paulovmr The problem with changed dropdown styles is actually not local to Paulo. I'm downloading fresh 6.4.x snapshot every morning and the problem appeared only yesterday.

cristianonicolai commented:
@paulovmr @jhrcek @manstis the style problem should be fixed by this PR: https://github.com/uberfire/uberfire-extensions/pull/178

paulovmr commented:
@cristianonicolai Thanks!

I updated the images on this PR's description with screenshots containing the correct style, and changed a little the behavior of the expand/collapse button.

Rikkola commented on kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/Explorer.java:
```diff
@@ -238,6 +211,58 @@ public void onClick( final ClickEvent clickEvent ) {
+        container.add( divClear );
+    }
+
+    private void setupNavigatorExpandCollapseButton() {
```
Good move on making this into it's own method. 
I have been lately making classes to reduce the over all class lengths. In this case you could make a NavigatorButton or NavigatorExpandCollapseButton and put everything from lines 221-225 and 239-243 there. Leaving just the buttton.addClickHandler(...) here.

paulovmr replied:
@Rikkola Good suggestion. I extended it a little and moved more stuff to this new class. Please, take a look and tell me what you think. :)

Rikkola replied:
Looks good. Now what the button is and how it functions is separated. 
Just wondering why the method name is build(), but it does not build anything?

paulovmr replied:
@Rikkola I agree, already changed it. How about now?

Rikkola commented on kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/navigator/NavigatorExpandCollapseButton.java:
```diff
@@ -0,0 +1,70 @@
+        initWidget( button );
+    }
+
+    public void build( final ClickHandler clickHandler ) {
```
@paulovmr This is the build method that does not build. Just to be clear.

csadilek commented:
+1

csadilek commented:
master: 1f0ca32d4804569715d6cda50eb167dc986f820f
